packages/suite/src/components/suite/Modal/index.tsx

```diff
@@ -47,35 +39,37 @@ const DefaultRenderer = ({
         [openGuide],
     );
 
-    return (
-        <GuideBackdrop
-            guideOpen={isGuideOpen && isModalOpen && !isGuideOnTop}
-            onClick={() => {
-                if (isCancelable) {
-                    onCancel?.();
-                }
-            }}
-        >
-            <TrezorModal
-                isCancelable={isCancelable}
-                onCancel={onCancel}
-                headerComponents={[...(isMobileLayout ? [GuideButton] : []), ...headerComponents]}
-                {...rest}
-            />
-        </GuideBackdrop>
-    );
-};
-
-export const Modal = ({ render: View = DefaultRenderer, ...props }: ModalProps) => {
-    const { isGuideOpen, isGuideOnTop } = useGuide();
+    if (!modalTarget) return null;
 
-    return (
+    const modal = (
         <FocusLock disabled={isGuideOpen && isGuideOnTop} group="overlay" autoFocus={false}>
-            <View {...props} />
+            <Backdrop
+                onClick={() => {
+                    if (isCancelable) {
+                        onCancel?.();
+                    }
+                }}
+            >
+                <TrezorModal
+                    isCancelable={isCancelable}
+                    onCancel={onCancel}
+                    headerComponents={[
+                        ...(isMobileLayout ? [GuideButton] : []),
+                        ...headerComponents,
+                    ]}
+                    {...rest}
+                />
+            </Backdrop>
         </FocusLock>
     );
+
+    return ReactDOM.createPortal(modal, modalTarget);
 };
 
+export const Modal = ({ render: View = DefaultRenderer, ...props }: ModalProps) => (
```

Comment:
Unfortunately yes, I would prefer to merge these changes and then possibly handle this. It would still require separation of content and modal-ish stuff in all the Confirm* and Pin* modals.

Reply:
Agree, it's a thing for another PR :)